libraries/lib-snapping/SnapUtils.cpp

```diff
@@ -0,0 +1,316 @@
+/*  SPDX-License-Identifier: GPL-2.0-or-later */
+/*!********************************************************************
+
+  Audacity: A Digital Audio Editor
+
+  SnapUtils.cpp
+
+  Dmitry Vedenko
+
+**********************************************************************/
+
+#include "SnapUtils.h"
+
+#include <algorithm>
+#include <cmath>
+#include <unordered_map>
+
+namespace
+{
+const wxString SnapModeKey = L"/Snap/Mode";
+const wxString SnapToKey = L"/Snap/To";
+const wxString OldSnapToKey = L"/SnapTo";
+const auto PathStart = L"SnapFunctions";
+
+struct RegistryVisitor : public Registry::Visitor
+{
```

Comment:
What you did here, to adapt one abstract visitor interface to another, might be more nicely done with a template convenience defined in Registry.h itself.

Reply:
... because one can easily foresee much repetition of the pattern